SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -361,113 +366,61 @@ void FLocalDeploymentManager::SetAutoDeploy(bool bInAutoDeploy)
 }
 
 // TODO: UNR-4423 Change this to taking a snapshot without the services
-void SPATIALGDKSERVICES_API FLocalDeploymentManager::TakeSnapshot(UWorld* World, bool bUseStandardRuntime,
-																  FSpatialSnapshotTakenFunc OnSnapshotTaken)
+void SPATIALGDKSERVICES_API FLocalDeploymentManager::TakeSnapshot(UWorld* World, FSpatialSnapshotTakenFunc OnSnapshotTaken)
 {
 	FHttpModule& HttpModule = FModuleManager::LoadModuleChecked<FHttpModule>("HTTP");
 	TSharedRef<IHttpRequest> HttpRequest = HttpModule.Get().CreateRequest();
 
-	HttpRequest->OnProcessRequestComplete().BindLambda(
-		[World, bUseStandardRuntime, OnSnapshotTaken](FHttpRequestPtr HttpRequest, FHttpResponsePtr HttpResponse, bool bSucceeded) {
-			if (!bSucceeded)
+	HttpRequest->OnProcessRequestComplete().BindLambda([World, OnSnapshotTaken](FHttpRequestPtr HttpRequest, FHttpResponsePtr HttpResponse,
+																				bool bSucceeded) {
+		if (!bSucceeded)
+		{
+			UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to trigger snapshot at '%s'; received '%s'"), *HttpRequest->GetURL(),
+				   *HttpResponse->GetContentAsString());
+			if (OnSnapshotTaken != nullptr)
 			{
-				UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to trigger snapshot at '%s'; received '%s'"),
-					   *HttpRequest->GetURL(), *HttpResponse->GetContentAsString());
+				OnSnapshotTaken(false /* bSuccess */, FString() /* PathToSnapshot */);
+			}
+			return;
+		}
+
+		// Unfortunately by the time this callback happens, the files haven't been flushed, so if you copy you may get
+		// the wrong info! So let's wait a bit..
+		FTimerHandle TimerHandle;
+		World->GetTimerManager().SetTimer(
+			TimerHandle,
+			[OnSnapshotTaken, HttpResponse]() {
+				bool bSuccess = false;
+
+				FString SnapshotSearchDirectory = SpatialGDKServicesConstants::SpatialOSSnapshotFolderPath;
+
+				FSpatialGDKServicesModule& GDKServices = FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+				FLocalDeploymentManager* LocalDeploymentManager = GDKServices.GetLocalDeploymentManager();
+
+				IFileManager& FileManager = FFileManagerGeneric::Get();
+
+				FString NewestSnapshotFilePath = HttpResponse->GetContentAsString();
+				FPaths::NormalizeFilename(NewestSnapshotFilePath);
+
+				bSuccess = FPaths::FileExists(NewestSnapshotFilePath);
+
+				if (!bSuccess)
+				{
+					UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed find snapshot file in '%s'"), *SnapshotSearchDirectory);
+				}
+
 				if (OnSnapshotTaken != nullptr)
 				{
-					OnSnapshotTaken(false /* bSuccess */, FString() /* PathToSnapshot */);
+					OnSnapshotTaken(bSuccess, NewestSnapshotFilePath);
 				}
-				return;
-			}
+			},
+			0.5f /* InRate */, false /* InbLoop */);
+	});
+
+	HttpRequest->SetURL(TEXT("http://localhost:5006/snapshot"));
+	HttpRequest->SetHeader("Content-Type", TEXT("application/json"));
```

Comment:
Probably not required, no content section in the request?

Reply:
Correct!